tests/test-prog.c

```diff
@@ -279,7 +296,9 @@ int main(int argc, char **argv)
     if (process_start(&proc, test_target) != 0) {
         return 1;
     }
-    printf("targe process started.\n");
+    is_musl = process_check_module(&proc, "/ld-musl-") == 1;
+    expected_errmsg = is_musl ? MUSL_INJECT_ERRMSG : INJECT_ERRMSG;
```

Comment:
1. `process_check_module` doesn't work as you expected. It returns `1` also for glibc processes.

2. Could you move the above two lines after `sleep(1)` below?
  At this point, `test_target` executed by `process_start` may be in the middle of initializing when this process go forward before CPU resources are assigned to the `test_target` process.  After `sleep(1`), `test_target` will be initialized.

Reply:
Done